File: calcularConstanciaDeLaFormaDelSonido.m
function [out_valorDeConstanciaDeLaFormaDelSonido] = calcularConstanciaDeLaFormaDelSonido(in_onda, in_frecuenciaDeMuestreo)
  
  longitudDeLaOnda = length(in_onda)
  
  unidadesDeTiempo = linspace(0, longitudDeLaOnda/in_frecuenciaDeMuestreo, longitudDeLaOnda);
  unidadesDeAmplitud = in_onda(1:longitudDeLaOnda);

  valorPromedio = sum(unidadesDeAmplitud)/longitudDeLaOnda
  umbral = valorPromedio*0.5
  limiteSuperior = valorPromedio + umbral
  limiteInferior = valorPromedio - umbral
  
  cantidadDeDatosEnElRango = 0;
  for dato = unidadesDeAmplitud
    if (limiteInferior <= dato && limiteSuperior >= dato)
      cantidadDeDatosEnElRango++;
    endif
  endfor
  cantidadDeDatosEnElRango
  (cantidadDeDatosEnElRango/longitudDeLaOnda) * 100
endfunction

File: Python/calcularConstanciaDeLaFormaDelSonido.py
import numpy

def calcularConstanciaDeLaFormaDelSonido(in_onda, in_frecuenciaDeMuestreo):

    print("---------Inicia calcularConstanciaDeLaFormaDelSonido ---------")
  
    longitudDeLaOnda = len(in_onda)

    in_onda /= max(abs(in_onda))
    
    limiteSuperior = max(in_onda)/2
    limiteInferior = min(in_onda)/2
      
    cantidadDeDatosEnElRango = 0
    for dato in in_onda:
        if (limiteInferior <= dato and limiteSuperior >= dato):
            cantidadDeDatosEnElRango += 1

    porcentajeDeValoresEnElRango = (cantidadDeDatosEnElRango/longitudDeLaOnda) * 100
    print("El límite superior es de ", limiteSuperior, ", el inferior de ", limiteInferior, " y el ", porcentajeDeValoresEnElRango, "% de los datos están en el rango")
    return porcentajeDeValoresEnElRango
